src/content/end.js

```diff
@@ -309,17 +319,12 @@ class Main extends Logged {
                 push = settings
             ] = Main.parseSettings(e[2]) || this.parseSubject(e[2], e[0]).concat(subjects)
             orders[order] = footer
-            let link = {
-                "BI-AAG": "https://courses.fit.cvut.cz/BI-AAG/",
-                "BI-AG1": "https://courses.fit.cvut.cz/BI-AG1/",
-                "BI-OSY": "https://courses.fit.cvut.cz/BI-OSY/",
-                "BI-PA1": "https://moodle-vyuka.cvut.cz/course/view.php?id=2203",
-                "BI-PA2": "https://courses.fit.cvut.cz/BI-PA2/",
-                "BI-PJV": "https://moodle-vyuka.cvut.cz/course/view.php?id=2265",
-                "BI-PS1": "https://courses.fit.cvut.cz/BI-PS1/",
-                "BI-PYT": "https://courses.fit.cvut.cz/BI-PYT/",
-                "unknown": "https://courses.fit.cvut.cz/",
-            }[e[2]]
+
+            let link = null
+            try {
+                link = subjectInfo["courses"][e[2]]["homepage"]
+            } catch (_) {}
```

Comment:
I don't like this very much. If fetching from courses fails we should either hide the subject link button or try to construct a fallback URL. Something like `https://courses.fit.cvut.cz/{subject name}`

Reply:
wait, my bad. I forgot that we already hide the button :D

Reply:
Implemented this in the newest commits.